pkg/scheduler/internal/cache/node_tree_test.go

```diff
@@ -428,14 +428,14 @@ func TestNodeTreeMultiOperations(t *testing.T) {
 		},
 		{
 			name:           "add more nodes to an exhausted zone",
-			nodesToAdd:     append(allNodes[4:9], allNodes[3]),
+			nodesToAdd:     append(append(make([]*v1.Node, 0), allNodes[4:9]...), allNodes[3]),
```

Comment:
you can also triple-slice the data to get a sub slice of bounded capacity so the append always allocates a new slice

Reply:
e.g. `allNodes[4:9:9]`

Reply:
Do you mean 'append(allNodes[4:9:9], allNodes[3])'? it doesn't seems work.

I did a test like this:
```
func main() {
    var original = [4]int{1, 2, 3, 4}
    copied := append(original[0:2:4], original[2])
    fmt.Printf("copied: %v\n", copied)
    copied[0] = 666
    fmt.Printf("original %v, copied: %v\n", original, copied)
}
```

The output is:
```
copied: [1 2 3]
original [666 2 3 4], copied: [666 2 3]
```

The orignal array still changed.

Reply:
Thanks!, I got it, it is more tidy. Will update soon